### PR TITLE
fix: 🐛 Various issues with syn-prio-nav

### LINKED ISSUE #409: fix: 🤔 syn-nav-item text is not centered if horizontal=true, prefix icon is available and text is too short
# Checklist (for the issue creator)

- [x] I have verified the bug exists in the latest version of `@synergy-design-system/components` (e.g. via `pnpm install @synergy-design-system/components`).
- [ ] I can provide a simple, reproducible example of the issue (e.g. via code-sandbox or sample code, ...).
- [ ] I would be willing to provide a PR for this issue.

# Version

The bug occurred with the following version of the component library: 1.15.0

# Component

The issue arises when using the following component(s): syn-nav-item

# Environment

The issue occurred with the following operating system(s): all
The issue occurred in the following browser(s): all

# Description
When using syn-nav-item with horizontal=true and with a prefix icon, it is possible that the text is not centered to the icon. Depending of the length of the text, the text is either centered (see left syn-nav-item of image) or it is not centered (see right syn-nav-item of image)
image

### Example code
```html
<syn-nav-item horizontal>
      <syn-icon name="wallpaper" slot="prefix" library="default"></syn-icon>
      abcdefghi
</syn-nav-item>

<syn-nav-item horizontal>
      <syn-icon name="wallpaper" slot="prefix" library="default"></syn-icon>
      abc
</syn-nav-item>

```

# Acceptance criteria

- [x] All DoD items successfully completed

- [x] the text is always centered to an icon, no matter how long the text is

### LINKED ISSUE #410: feat: ✨ SynPrioNav: Adjust syn-prio-nav shrinking mechanism
# Checklist (for the issue creator)

- [x] I have looked at the documentation site and was not able to find the required feature.
- [ ] I have looked into the Figma boards and was not able to find the required feature.
- [ ] I would be willing to provide a PR for this feature.

# Component

I suggest to add the feature to the following component: syn-prio-nav

# Story

Currently the syn-prio-nav shows the priority menu only, if there is no space left for any navigation item. If there is space for one navigation item, it shows this navigation item. See following image for the current behavior:
image

From UX perspective the priority menu only should also be shown if there is only one space left for one navigation item. So in the image above the first syn-prio-nav should look the same like the second with the priority menu only.

# Acceptance criteria

- [x] All DoD items successfully completed

- [x] syn-prio-nav shows priority menu only, if there is space for just one navigation item left
- [x] docs are updated in doc tokens about the behavior

### LINKED ISSUE #582: fix: 🐛 SynSideNav: Parent nav item retains the selected state until being expanded
# Checklist (for the issue creator)

- [ ] I have verified the bug exists in the latest version of `@synergy-design-system/components` (e.g. via `pnpm install @synergy-design-system/components`).
- [ ] I can provide a simple, reproducible example of the issue (e.g. via code-sandbox or sample code, ...).
- [ ] I would be willing to provide a PR for this issue.

# Version

The bug occurred with the following version of the component library: 2.4.1

# Component

The issue arises when using the following component(s): syn-side-nav

# Environment

The issue occurred with the following operating system(s): all
The issue occurred in the following browser(s): all

# Description

Using syn-side-nav and configure nested nav items in the application, when user clicked on the nested nav item and collapsed the child nav item, the parent nav item is highlighted. The parent nav item will not be de-selected even if user click on other nav item. Until user expand the child nav items.

https://github.com/user-attachments/assets/e83ad408-7af4-4c48-ace7-f26d8903a887

# Acceptance criteria

- [ ] All DoD items successfully completed

### LINKED ISSUE #630: fix: 🐛 menu text in <syn-prio-nav> is not aligned with baseline of other text
# Checklist (for the issue creator)

- [x] I have verified the bug exists in the latest version of `@synergy-design-system/components` (e.g. via `pnpm install @synergy-design-system/components`).
- [x] I can provide a simple, reproducible example of the issue (e.g. via code-sandbox or sample code, ...).
- [ ] I would be willing to provide a PR for this issue.

# Version

The bug occurred with the following version of the component library: 2.9.0

# Component

The issue arises when using the following component(s): syn-prio-nav

# Environment

The issue occurred with the following operating system(s): all
The issue occurred in the following browser(s): all

# Description

https://codepen.io/xkabylgSICKAG/pen/qBeBreZ?editors=1110

image

This can be fixed by setting `display: flex; align-items: center;` on the `::part(content-container)` of syn-nav-item, but the syn-nav-item inside the prio-nav isn't accessible to our applications.

# Acceptance criteria

- [ ] All DoD items successfully completed
- [ ] All text in the priority menu share a common baseline

### LINKED ISSUE #631: fix: 🐛 SynPrioNav: adding href to <syn-nav-item> in  <syn-prio-nav> results in bad text alignment
# Checklist (for the issue creator)

- [x] I have verified the bug exists in the latest version of `@synergy-design-system/components` (e.g. via `pnpm install @synergy-design-system/components`).
- [x] I can provide a simple, reproducible example of the issue (e.g. via code-sandbox or sample code, ...).
- [ ] I would be willing to provide a PR for this issue.

# Version

The bug occurred with the following version of the component library: 2.9.0

# Component

The issue arises when using the following component(s): syn-prio-nav, syn-nav-item

# Environment

The issue occurred with the following operating system(s): all
The issue occurred in the following browser(s): all

# Description

image
https://codepen.io/xkabylgSICKAG/pen/MWNWmog

This can be fixed by adding the following css:

```css
syn-nav-item::part(base) {
  display: flex;
  align-items: center;
}
```

# Acceptance criteria

- [ ] All DoD items successfully completed
- [ ] mixing buttons and links in a prio nav results in the same baseline alignment

### LINKED ISSUE #639: fix: 🐛 syn-prio-nav: priority menu position is always calculated from the absolute left of the document
# Checklist (for the issue creator)

- [x] I have verified the bug exists in the latest version of `@synergy-design-system/components` (e.g. via `pnpm install @synergy-design-system/components`).
- [x] I can provide a simple, reproducible example of the issue (e.g. via code-sandbox or sample code, ...).
- [x] I would be willing to provide a PR for this issue.

# Version

The bug occurred with the following version of the component library: >= 2.0.0

# Component

The issue arises when using the following component(s): syn-prio-nav

# Environment

The issue occurred with the following operating system(s): All
The issue occurred in the following browser(s): All

# Description

The `<syn-prio-nav>` is able to hide all items that may not be visible on the screen in its own priority menu, as seen in https://synergy-design-system.github.io/?path=/docs/components-syn-prio-nav--docs#priority%20menu. However, the calculation is always taken from the top left position of the screen, making it unusable if the navigation is placed somewhere inside a document (e.g. by applying a padding to it).

This was originally found and requested by @xkabylgSICKAG.

Examples: https://codepen.io/xkabylgSICKAG/pen/YzoMjaQ
Original PR: https://github.com/synergy-design-system/synergy-design-system/pull/608

# Acceptance criteria

- [ ] All DoD items successfully completed
- [ ] The priority menu should work whereever the `<syn-prio-nav>` is placed.

### LINKED ISSUE #644: fix: 🐛 Umbrella: SynPrioNav
# Checklist (for the issue creator)

- [x] I have verified the bug exists in the latest version of `@synergy-design-system/components` (e.g. via `pnpm install @synergy-design-system/components`).
- [x] I can provide a simple, reproducible example of the issue (e.g. via code-sandbox or sample code, ...).
- [x] I would be willing to provide a PR for this issue.

# Version

The bug occurred with the following version of the component library: 2.9.0

# Component

The issue arises when using the following component(s): syn-prio-nav, syn-nav-item

# Description

This ticket acts as an umbrella ticket for all currently open problems that happen on `<syn-prio-nav>` and `<syn-nav-item>`

# Acceptance criteria

- [x] All DoD items successfully completed
- [x] All of the following tickets are fixed and tested in the branch:
- [x] #582
- [x] #631
- [x] #410
- [x] #409
- [x] #639

### PR DESCRIPTION
**Pull request by schilchSICKAG**
# Pull Request

## 📖 Description

This PR acts as an umbrella for various tickets related to the `<syn-prio-nav>` and `<syn-nav-item>` components.

### 🎫 Issues

Closes #644 
Closes #582
Closes #631
Closes #410
Closes #409
Closes #630 
Closes #639

## 👩‍💻 Reviewer Notes

This PR is used to add multiple fixes to our navigation items. It is easier to review those tickets in one scope, so we created an umbrella ticket housing all changes.

## 📑 Test Plan

The following list of steps describes all issues and their related tests:

### Issue #631

Before: https://codepen.io/schilchSICKAG/pen/LYwGraY?editors=1110
After: http://localhost:6006/?path=/docs/components-syn-prio-nav--docs

tl;dr: When using a href attribute, the alignment must now always fit in horizontal mode. The default story of `<syn-prio-nav>` was adjusted to include a `<syn-nav-item>` with a href attribute to check this also in visual integration.

---

### Issue #410

When displaying a `<syn-prio-nav>` where only one item would be visible, this item is now also hidden. Play around with the priority-menu story for this to test.

Open question: I have currently the marker internally in the method:

```typescript
private handlePriorityMenu() {
  // code
  const visibleItems = itemVisibilities.filter(({ isHidden }) => !isHidden).length;
  const hasOnlyOneItem = visibleItems === 1;
  // even more code :(
}
```

It would be possible to let users control this behavior if we would use an attribute for this (e.g. something like `<syn-prio-nav items-before-menu-visible="2"></syn-prio-nav>`. Anyone seeing any value in this?

---

### Issue #409

~~The fix was archived by making sure to apply the styling only if the `<syn-nav-item>` is not in horizontal mode (e.g. `.nav-item--show-prefix-only .nav-item__content-container` became `:host(:not([horizontal])) .nav-item--show-prefix-only .nav-item__content-container`.~~

**Note**: We could also have adjusted the code to make sure the `handleWidth` is only called when the `horizontal` attribute is set. From a code perspective, this would have various advantages, like calling the listener only when NOT in horizontal mode. If wanted, we could do this easily, too by adding a `@watch` decorator to the `horizontal` and pausing / starting the `ResizeObserver` if it is true. Let´s discuss this. The current solution is working fine either way.

> We have opted to use the code centric above. The resizeObserver is now paused when the `horizontal` prop is detected (or in case of mounting the component, not even started when it is set).

---

### Issue #582 

Before: https://synergy-design-system.github.io/?path=/docs/components-syn-side-nav--docs#indentation
After: http://localhost:6006/?path=/docs/components-syn-side-nav--docs#indentation

This issue was hard to fix because we do not have information about the structure of a navigation. The only thing we really know are the children of a given `<syn-nav-item>`. I have now opted to remove the original `@watch` decorator for the open property as there is no way to trigger this correctly when the user clicks another item. However, what CAN be done is listening for changes to the `open` and `current` props, together with a `MutationObserver`. With this, it will happily apply the changes. There are some minor performance problems on this (e.g. the change list will also include all sub-parts), but I think this is a minor problem and did not lead to bigger performance issues.

## ✅ DoD

- [x] I have tested my changes manually.
- [x] I have added automatic tests for my changes (unit- and visual regression tests).
- [x] I have updated the project documentation to reflect my changes (e.g. CHANGELOG, Storybook, ...).
- [x] I have added documentation to the DaVinci migration guide (if need be)
- [x] I have made sure to follow the projects coding and contribution guides.
- [x] I have made sure that the bundle size has changed appropriately.
- [x] I have validated that there are no accessibility errors.
- [x] I have used design tokens instead of fix css values

### CODE REVIEW

**Comment by kirchsuSICKAG** [comment]
I just checked it, following issue is also fixed with this PR: #630

**Comment by kirchsuSICKAG** [comment]
> ### Issue #410
> Open question: I have currently the marker internally in the method:
>
>It would be possible to let users control this behavior if we would use an attribute for this (e.g. something like <syn-prio->nav items-before-menu-visible="2"></syn-prio-nav>. Anyone seeing any value in this?

From my feeling I would say no, I don't really see a value. 
For the moment I would leave it as it is. If someone needs the feature / asks for it, we could add it later.

**Comment by kirchsuSICKAG** [comment]
> ### Issue #409
>
>Note: We could also have adjusted the code to make sure the handleWidth is only called when the horizontal attribute is set. From a code perspective, this would have various advantages, like calling the listener only when NOT in horizontal mode. If wanted, we could do this easily, too by adding a @watch decorator to the horizontal and pausing / starting the ResizeObserver if it is true. Let´s discuss this. The current solution is working fine either way.

I like this proposal! It makes it clear in the code, that the multiline and the prefix-only styling is only needed, if `horizontal=false`.

**Review by kirchsuSICKAG — CHANGES_REQUESTED** [review]

**Comment by schilchSICKAG** [comment]
> > ### Issue #409
> > Note: We could also have adjusted the code to make sure the handleWidth is only called when the horizontal attribute is set. From a code perspective, this would have various advantages, like calling the listener only when NOT in horizontal mode. If wanted, we could do this easily, too by adding a @watch decorator to the horizontal and pausing / starting the ResizeObserver if it is true. Let´s discuss this. The current solution is working fine either way.
> 
> I like this proposal! It makes it clear in the code, that the multiline and the prefix-only styling is only needed, if `horizontal=false`.

Done in the latest version

**Comment by schilchSICKAG** [comment]
> > ### Issue #409
> > Note: We could also have adjusted the code to make sure the handleWidth is only called when the horizontal attribute is set. From a code perspective, this would have various advantages, like calling the listener only when NOT in horizontal mode. If wanted, we could do this easily, too by adding a @watch decorator to the horizontal and pausing / starting the ResizeObserver if it is true. Let´s discuss this. The current solution is working fine either way.
> 
> I like this proposal! It makes it clear in the code, that the multiline and the prefix-only styling is only needed, if `horizontal=false`.

This is now fixed

File: packages/components/src/components/nav-item/nav-item.styles.ts [hunk 1]
```diff
@@ -203,13 +203,13 @@ export default css`
...
-  .nav-item--has-prefix.nav-item--has-suffix .nav-item__content-container {
+  :host(:not([horizontal])) .nav-item--has-prefix.nav-item--has-content .nav-item__content-container,
+  :host(:not([horizontal])) .nav-item--has-prefix.nav-item--has-suffix .nav-item__content-container {
     margin-inline-start: var(--syn-spacing-x-small);
   }
 
-  .nav-item--has-suffix.nav-item--has-content .nav-item__content-container,
-  .nav-item--has-suffix.nav-item--has-prefix .nav-item__content-container {
+  :host(:not([horizontal])) .nav-item--has-suffix.nav-item--has-content .nav-item__content-container,
+  :host(:not([horizontal])) .nav-item--has-suffix.nav-item--has-prefix .nav-item__content-container {
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/components/nav-item/nav-item.styles.ts]: Should be resolved with the changes to the underlying resizeObserver

**Review by kirchsuSICKAG — APPROVED** [review]
